crates/test_utils/src/connector_auth.rs

```diff
@@ -0,0 +1,251 @@
+use std::{collections::HashMap, env};
+
+use router::types::ConnectorAuthType;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ConnectorAuthentication {
+    pub aci: Option<BodyKey>,
+    pub adyen: Option<BodyKey>,
+    pub adyen_uk: Option<BodyKey>,
+    pub airwallex: Option<BodyKey>,
+    pub authorizedotnet: Option<BodyKey>,
+    pub bambora: Option<BodyKey>,
+    pub bitpay: Option<HeaderKey>,
+    pub bluesnap: Option<BodyKey>,
+    pub cashtocode: Option<BodyKey>,
+    pub checkout: Option<SignatureKey>,
+    pub coinbase: Option<HeaderKey>,
+    pub cryptopay: Option<BodyKey>,
+    pub cybersource: Option<SignatureKey>,
+    pub dlocal: Option<SignatureKey>,
+    #[cfg(feature = "dummy_connector")]
+    pub dummyconnector: Option<HeaderKey>,
+    pub fiserv: Option<SignatureKey>,
+    pub forte: Option<MultiAuthKey>,
+    pub globalpay: Option<BodyKey>,
+    pub iatapay: Option<SignatureKey>,
+    pub mollie: Option<HeaderKey>,
+    pub multisafepay: Option<HeaderKey>,
+    pub nexinets: Option<BodyKey>,
+    pub noon: Option<SignatureKey>,
+    pub nmi: Option<HeaderKey>,
+    pub nuvei: Option<SignatureKey>,
+    pub opayo: Option<HeaderKey>,
+    pub opennode: Option<HeaderKey>,
+    pub payeezy: Option<SignatureKey>,
+    pub payme: Option<HeaderKey>,
+    pub paypal: Option<BodyKey>,
+    pub payu: Option<BodyKey>,
+    pub rapyd: Option<BodyKey>,
+    pub shift4: Option<HeaderKey>,
+    pub stripe: Option<HeaderKey>,
+    pub stripe_au: Option<HeaderKey>,
+    pub stripe_uk: Option<HeaderKey>,
+    pub trustpay: Option<SignatureKey>,
+    pub worldpay: Option<BodyKey>,
+    pub worldline: Option<SignatureKey>,
+    pub zen: Option<HeaderKey>,
+    pub automation_configs: Option<AutomationConfigs>,
+}
+
+impl Default for ConnectorAuthentication {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[allow(dead_code)]
+impl ConnectorAuthentication {
+    #[allow(clippy::expect_used)]
+    pub fn new() -> Self {
+        // Do `export CONNECTOR_AUTH_FILE_PATH="/hyperswitch/crates/router/tests/connectors/sample_auth.toml"`
+        // before running tests in shell
+        let path = env::var("CONNECTOR_AUTH_FILE_PATH")
+            .expect("Connector authentication file path not set");
+        toml::from_str(
+            &std::fs::read_to_string(path).expect("connector authentication config file not found"),
+        )
+        .expect("Failed to read connector authentication config file")
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct ConnectorAuthenticationMap(pub HashMap<String, ConnectorAuthType>);
```

Comment:
Can you add an `inner()` method that returns a reference to the inner `HashMap`, and make the tuple field private?